### DISCUSSION TITLE
Requires amd64 servers

### DISCUSSION THREAD

**Discussion by mrconnerton-encorehc** [Tab-SE/BridgeCTL#10] [General]
Should probably specific in the quickstart / install that building the containers require amd64 host machine. I tried on a t4g arm64 and it failed to build images.

**Comment by sf-jfluckiger** [comment]
Good feedback. We'll updated the docs. @matusb could you add this?

**Comment by matusb** [comment]
I will test it and update the documentation. Thank you

**Comment by matusb** [comment]
Looking for Mac with arm64 to test following 
```docker run --rm -ti --platform linux/amd64 ubuntu:latest uname -m x86_64```

**Comment by sf-jfluckiger** [comment]
I added a System Test tab to the Settings page. Does this look good?
<img width="1268" alt="image" src="https://github.com/user-attachments/assets/f4b9a6ea-b34c-4f32-b56a-71ccf6795874">


Another idea is that upon startup we show a welcome screen. And this welcome screen can show a "System Check" button which will run the command to see if their system has the required components.
(docker, SSL in python configured correclty, ability to build amc64 images in docker )

We only show this screen once and the user can bypass the welcome screen if they'd like (they might only be using the log viewer and so don't care about docker). If they use wants to run the system check again after startup, they can do so from the Settings tab at any time.

I would modify your command slightly to use rhel9 since that is what most customers use. But yes, I agree this is a good way to test their system.
docker run --rm -ti --platform linux/amd64 registry.hub.docker.com/redhat/ubi9:latest uname -m

**Comment by matusb** [comment]
Can we use the platform flag for building/running containers by default? It should not affect amd64 hosts, but should address issues on M1/M2 chipsets.

**Comment by josephflu** [comment]
We already pass in this flag when building the image. But there may be a difference between using the python docker library to build an image  and calling docker build from the command-line. I will test this more.
